src/client/api/object.c

```diff
@@ -165,7 +165,7 @@ daos_obj_query_key(daos_handle_t oh, daos_handle_t th, uint32_t flags,
 int
 daos_obj_fetch(daos_handle_t oh, daos_handle_t th, daos_key_t *dkey,
 	       unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
-	       daos_iom_t *maps, daos_event_t *ev)
+	       struct daos_io_map *maps, daos_event_t *ev)
```

Comment:
i think we should use the typedef'd daos_iom_t.
otherwise the API looks weird with daos_iod_t, and then struct daos_io_map

Reply:
our plan is removing most of typedefs gradually (we will keep typedef for small structures like object ID, d_iov_t, but remove typedefs for data structures which are large and not suitable for putting in function call stack).

So I'm keeping both structure name and typedef name for some structures, we can use structure name in new code, and removing typedefed name incrementally, instead of generating another a few thousands of lines patch which renames everything.